src/rules/unbound-method.ts

```diff
@@ -0,0 +1,111 @@
+import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import { createRule, isExpectCall, parseExpectCall } from './utils';
+
+const toThrowMatchers = [
+  'toThrow',
+  'toThrowError',
+  'toThrowErrorMatchingSnapshot',
+  'toThrowErrorMatchingInlineSnapshot',
+];
+
+const isJestExpectCall = (node: TSESTree.CallExpression) => {
+  if (!isExpectCall(node)) {
+    return false;
+  }
+
+  const { matcher } = parseExpectCall(node);
+
+  if (!matcher) {
+    return false;
+  }
+
+  return !toThrowMatchers.includes(matcher.name);
+};
+
+const baseRule = (() => {
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const TSESLintPlugin = require('@typescript-eslint/eslint-plugin');
+
+    return TSESLintPlugin.rules['unbound-method'] as TSESLint.RuleModule<
+      MessageIds,
+      Options
+    >;
+  } catch (e: unknown) {
+    const error = e as { code: string };
+
+    if (error.code === 'MODULE_NOT_FOUND') {
+      return null;
+    }
+
+    throw error;
+  }
+})();
+
+const tryCreateBaseRule = (
+  context: Readonly<TSESLint.RuleContext<MessageIds, Options>>,
+) => {
+  try {
+    return baseRule?.create(context);
```

Comment:
what errors can this throw?

Reply:
I don't know, but figured it'd be best to play it safe.

The only error I expect it will throw is that if you don't provide `project`, it'll throw an error about not having the type checking service.